Doc updates for storage (#1570)
openmls/openmls: book/src/traits/traits.md

{{#include ../../../traits/src/crypto.rs:10}}
```

### StorageProvider

This trait defines an API for a storage backend that is used for all OpenMLS
persistence.

The store provides functions to `store`, `read`, and `delete` values.
Note that it does not allow updating values.
Instead, entries must be deleted and newly stored.

```rust,no_run,noplayground
{{#include ../../../traits/src/storage.rs:16:25}}
```

The trait is generic over a `VERSION`, which is used to ensure that the values
that are persisted can be upgraded when OpenMLS changes the stored structs.

Every function takes `Key` and `Value` arguments.

```rust,no_run,noplayground
{{#include ../../../traits/src/storage.rs:key_trait}}
```

```rust,no_run,noplayground
{{#include ../../../traits/src/storage.rs:entity_trait}}
```

To ensure that each function takes the correct input, they use trait bounds.
These are the available traits.

```rust,no_run,noplayground
{{#include ../../../traits/src/storage.rs:traits}}
```

An implementation of the storage trait should ensure that it can address and
efficiently handle values.

#### Example: Key packages

> This is only an example, but illustrate that the application may need to do more

Reviewer: ```suggestion
This is only an example, but it illustrates that the application may need to do more
```

when it comes to implementing storage.

Key packages are only deleted by OpenMLS when they are used and _not_ last resort

> key packages (which are supposed to be used multiple times).

Reviewer: ```suggestion
key packages (which may be used multiple times).
```

The application needs to implement some logic to manage last resort key packages.

```rust,no_run,noplayground
{{#include ../../../traits/src/storage.rs:write_key_package}}
```

> The application may store the has references in a separate list with a validity

Reviewer: ```suggestion
The application may store the hash references in a separate list with a validity
```
